samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/DateResolverDialog.java

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.core;
+
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.dialogs.DialogTurnResult;
+import com.microsoft.bot.dialogs.WaterfallDialog;
+import com.microsoft.bot.dialogs.WaterfallStep;
+import com.microsoft.bot.dialogs.WaterfallStepContext;
+import com.microsoft.bot.dialogs.prompts.DateTimePrompt;
+import com.microsoft.bot.dialogs.prompts.DateTimeResolution;
+import com.microsoft.bot.dialogs.prompts.PromptOptions;
+import com.microsoft.bot.dialogs.prompts.PromptValidatorContext;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.InputHints;
+import com.microsoft.recognizers.datatypes.timex.expression.Constants;
+import com.microsoft.recognizers.datatypes.timex.expression.TimexProperty;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ *
+ */
+public class DateResolverDialog extends CancelAndHelpDialog {
+    private final String promptMsgText = "When would you like to travel?";
+    private final String repromptMsgText =
+        "I'm sorry, to make your booking please enter a full travel date including Day Month and Year.";
+
+    /**
+     *
+     * @param id
+     */
+    public DateResolverDialog(@Nullable String id) {
+        super(id != null ? id : DateResolverDialog.class.getName());
+
+        addDialog(new DateTimePrompt(DateTimePrompt.class.getName(), DateResolverDialog::dateTimePromptValidator));
+        WaterfallStep[] waterfallSteps = {
+            this::initialStep,
+            this::finalStep
+        };
+        addDialog(new WaterfallDialog(WaterfallDialog.class.getName(), Arrays.asList(waterfallSteps)));
+
+        setInitialDialogId(WaterfallDialog.class.getName());
+    }
+
+    private CompletableFuture<DialogTurnResult> initialStep(WaterfallStepContext stepContext) {
+        String timex = stepContext.getOptions().toString();
+
+        Activity promptMessage = MessageFactory.text(promptMsgText, promptMsgText, InputHints.EXPECTING_INPUT);
+        Activity repromptMessage = MessageFactory.text(repromptMsgText, repromptMsgText, InputHints.EXPECTING_INPUT);
+
+        if (StringUtils.isBlank(timex)) {
```

Comment:
```suggestion
        if (timex == null) {
```
[C#](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/13.core-bot/Dialogs/DateResolverDialog.cs#L40) checks for null